internal/flush/flush_coordinator.go

```diff
@@ -0,0 +1,55 @@
+package flush
+
+type Coordinator interface {
```

Comment:
I would rename this to `FlushNotifier` instead of `Coordinator` since it does extend `Flushable`

Reply:
The only problem is there's already a something called `flushNotifier` in the repo, I made it a coordinator because you can also invoke a flush manually

The other flushNotifier interface is strictly for notification of flushes